Engine/Component/Component.h

```diff
@@ -79,6 +80,9 @@ class Component
 	virtual void SpecializedSave(Config& config) const = 0;
 	virtual void SpecializedLoad(const Config &config) = 0;
 
+	virtual void GenerateTextures(TextureLoadData loaded_data) {}
```

Comment:
I guess you are creating this as a starting point but would be more generic in the future? like GenerateResource

Reply:
The thing is for other resources it needs other structures but yeah I could do different declarations of GenerateResource with the different parameters

Reply:
You can remove this one now right?